compiler/parser/RdrHsSyn.hs

```diff
@@ -2602,8 +2634,13 @@ mkTemplateInstanceDecl templateName conName ValidTemplate{..} =
     mkMethod :: String -> [Pat GhcPs] -> Bool -> LHsExpr GhcPs -> LHsBind GhcPs
     mkMethod methodName args includeBindings methodBody =
       mkTemplateClassMethod methodName args methodBody $
-          -- General rule: only include template bindings for methods with `this` in scope
-          if includeBindings then Just vtLetBindings else Nothing
+          -- General rule: only include template bindings for methods
+          -- with `this` in scope.
+          if not includeBindings then
```

Comment:
Maybe worth factoring out given that you duplicate this below?

Reply:
✔️